### PR TITLE
feat(self-managed): restructure and add docs for new backup api in Zeebe

### LINKED ISSUE #1536: Document Zeebe S3 backup store `basePath` config
Depends on https://github.com/camunda/zeebe/pull/11108

### PR DESCRIPTION
**Pull request by deepthidevaki**
## What is the purpose of the change

- Restructured zeebe backup docs
   - Moved configuration from end to the beginning (Prerequisite)
   - Renamed heading of each api to match that of Optimize docs
   - Each api has a Request and Response section
- Added docs for list and delete api
- Updated docs for create backup and get backup info
- This PR documents the final api for Zeebe backups. So note saying the api is subject to change is also removed.

The docs matches with the [openapi spec](https://github.com/camunda/zeebe/blob/main/dist/src/main/resources/api/backup-management-api.yaml) available in zeebe repo. 

Related to https://github.com/camunda/zeebe/issues/10809
Closes #1536 

## Are there related marketing activities

No

## When should this change go live?

Release 8.2.0

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@npepinpe I'm assigning you for reviewing this so that you get a chance to review the latest api.

**Comment by deepthidevaki** [comment]
Since there is a lot of change. It might be easier to review the whole page instead of the diff.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 1]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+The API is accessible via the management port of the gateway.
+
+## Prerequisites
 
-A backup of a Zeebe cluster consists of a consistent snapshot of all partitions. The backup is stored externally. The backup is taken asynchronously in the background while Zeebe is processing. Thus, the backups can be taken with minimal impact on normal processing.
+To use the backup feature in Zeebe, the following configurations must be provided.
 
-Backup of a running Zeebe cluster can be taken using the following rest APIs. The API is accessible via the management port of the gateway.
+- Enable backups by setting the flag `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEBACKUP` to `true`.
+- The backup management API is available via management port of the gateway. Ensure the configuration `MANAGEMENT_ENDPOINTS_WEB_EXPOSURE_INCLUDE` include "backups". Set `MANAGEMENT_ENDPOINTS_BACKUPS_ENABLED` to `true`.
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:21]: :wrench: I don't know if everyone is familiar with Spring Boot configuration, so I would add a note that the backup management API is configurable as an Spring Boot actuator endpoint (with a link to how to configure those), and its ID is `backups`. This would be complimentary to what you already added, of course.

Not sure what's better, if you add it here, or if you add it as a `:::note` paragraph below :thinking:

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 2]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+- Backup is stored in an external storage. This must be configured before starting the Zeebe cluster.
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:22]: :upside_down_face: I would split this away from the prerequisites, or at least make it a sub-step (with its own, lower-level, header), specifically around configuring the external backup store. Soft suggestion, but I think it's a slightly more "complex" point than the previous ones, and it sets up a separation for further storage plugins in the future.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 3]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+            endpoint:
+            accessKey:
+            secretKey:
+```
+
+Alternatively, you can configure backup store using environment variables:
 
-- [Trigger backup](#trigger-backup)
-- [Monitor backup](#monitor-backup)
+- `ZEEBE_BROKER_DATA_BACKUP_STORE` - Specify which storage to use as the backup storage. Currently, only S3 is supported. You can use any S3-compatible storage.
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:43]: :upside_down_face: I think it's easy to miss the "Currently, only S3 is supported". Maybe we can make it more explicit by bolding it and already writing about it earlier in the intro. I have a feeling it'll be easy for people to gloss over here, but maybe not :shrug:

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 4]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+- `ZEEBE_BROKER_DATA_BACKUP_S3_BUCKETNAME` - The backup will be stored in this bucket. The bucket must already exist.
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:44]:
```suggestion
- `ZEEBE_BROKER_DATA_BACKUP_S3_BUCKETNAME` - The backup will be stored in this bucket. **The bucket must already exist.**
```

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 5]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+- `ZEEBE_BROKER_DATA_BACKUP_S3_BASEPATH` - If the bucket is shared with other Zeebe clusters, then a unique basePath must be configured.
+- `ZEEBE_BROKER_DATA_BACKUP_S3_ENDPOINT` - If no endpoint is provided, it will be determined based on the configured region.
+- `ZEEBE_BROKER_DATA_BACKUP_S3_REGION` - If no region is provided, it will be determined [from the environment](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html#automatically-determine-the-aws-region-from-the-environment).
+- `ZEEBE_BROKER_DATA_BACKUP_S3_ACCESSKEY` - If either `accessKey` or `secretKey` is not provided, the credentials will be determined [from the environment](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html#credentials-chain).
+- `ZEEBE_BROKER_DATA_BACKUP_S3_SECRETKEY`
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:49]: :question: No docs? :sweat_smile: I guess it's sort of bundled with the above? :shrug:

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 6]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+```
+curl --request GET 'http://localhost:9600/actuator/backups'
+```
+
+</details>
+
+### Response
+
+| Code             | Description                                                                                |
+| ---------------- | ------------------------------------------------------------------------------------------ | --- |
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:197]: :question: Why three columns here?

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 7]
```diff
@@ -5,110 +5,279 @@ description: "A guide to creating backup of a running Zeebe cluster."
...
+```
+curl --request DELETE 'http://localhost:9600/actuator/backups/100'
+```
+
+</details>
+
+### Response
+
+| Code             | Description                                                                      |
+| ---------------- | -------------------------------------------------------------------------------- | --- |
```
Comment (npepinpe) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:275]: :question: Why 3 columns?

**Review by npepinpe — DISMISSED** [review]
:rocket: 

Only some suggestions, no blockers. Looks very cool now! I hope we use OpenAPI more often for management stuff :)

- :wrench: Could we provide a link to the OpenAPI spec directly? That way users can easily generate stubs from it. Or could we even embed/add it here? I wonder :thinking: Or let me rephrase, what would be great is a way to make it as easy as possible for users to reuse OpenAPI tooling based on our spec :+1:

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md (hunk 1 above)
Comment (deepthidevaki) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md]: Added it to the intro and referred to spring boot docs.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md (hunk 3 above)
Comment (deepthidevaki) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md:50]: :+1: Explained it in intro.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md (hunk 6 above)
Comment (deepthidevaki) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md]: Fixed!

**Comment by deepthidevaki** [comment]
Thanks @npepinpe for the review. I have applied your suggestions.

>  Could we provide a link to the OpenAPI spec directly?

I have now added a link to the openapi spec in zeebe repo main branch. This would be good enough for now. But I have no idea how to handle versions, if the api gets changed.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md [hunk 8]
```diff
@@ -1,114 +1,285 @@
...
+The backups are stored to an external data storage. [S3](https://aws.amazon.com/s3/) or any S3-compatible storage is supported as the backup storage.
+
+## Prerequisites
+
+To use the backup feature in Zeebe, the following configurations must be provided:
 
-A backup of a Zeebe cluster consists of a consistent snapshot of all partitions. The backup is stored externally. The backup is taken asynchronously in the background while Zeebe is processing. Thus, the backups can be taken with minimal impact on normal processing.
+- Enable backups by setting the flag `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEBACKUP` to `true`.
+- Ensure the configuration `MANAGEMENT_ENDPOINTS_WEB_EXPOSURE_INCLUDE` includes the endpoint "backups". Set `MANAGEMENT_ENDPOINTS_BACKUPS_ENABLED` to `true`.
+  For additional configurations such as security, refer to the Spring Boot documentation.
```
Comment (christinaausley) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:21]: Could you provide a link to this here?

**Review by christinaausley — APPROVED** [review]
Technical adjustments made in my latest commit for some clarity, punctuation, and grammar. Let me know if you have any questions about these 👍 

One non-blocking comment as well.

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md (hunk 8 above)
Comment (deepthidevaki) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md]: Added link to spring boot docs.

**Comment by deepthidevaki** [comment]
@christinaausley Could you approve again so I can merge it?

**Review by christinaausley — APPROVED** [review]
